### ISSUE TITLE
Unable to invade populated worlds

### ISSUE THREAD

**Issue by MatGB** [freeorion/freeorion#1778]
* **FreeOrion Version**: f5e26bc (with irrelevent script changes)
* **Operating System**: Kubuntu 16.04
  * Compiled from source

### Description
I'm unable to invade some AI worlds, it appears to be only inhabited worlds and only after Planetary Barrier Shield 1 has been researched, a debug data dump shows the worlds that are displaying 0 shields actually have 0.25 shields which is enough to stop invasion.

The AI I'm specifically attacking completed PBS1 turn 125 (useful info Geoff, thanks), the one I gave up on a few turns earlier had completed the tech on turn 89: I'm able to invade outposts which is what gave me the right clue.

### Expected Result
A planet incapacitated in a combat should be invadable the following turn, always.

### Steps to reproduce
[invasion-blocked-save.zip](https://github.com/freeorion/freeorion/files/1355054/invasion-blocked-save.zip)
First save click on Jinga a system or Saule b, second save Mier b.

(I'm going to hack the FOCS files but I can't see where the change has come from)

**Comment by MatGB** [comment]
I've tried various FOCS hacks including the CANDIDATE_BATTLE_CHECK macro combined with an infrastructure reduction and nothing has quite worked, if this is affecting others (which it ought to be given my script changes are entirely research tech tree organisation) then invasions become impossible once that tech is researched and I cannot see why.

During turn processing the 'invade' option flashes up between two game state loads (after combat) but you can't use it (obviously) and it goes again with the next update.

**Comment by dbenage-cx** [comment]
Can not reproduce with either save (guessing you meant Meir y here):
![screenshot_20171005_142704](https://user-images.githubusercontent.com/17205519/31249580-83162972-a9dd-11e7-89c0-e899ac78d6e1.png)

On load of save from turn 129, there is an AI exception, but doubt it is relevant as the planet shows accounting for PBS1:
```
AI_4: AI_Error: AI script error : FreeOrionAI.py:generateOrders():333  - Exception 114698 while trying to assign_military_fleets_to_systems
Traceback (most recent call last):
  File "/opt/projects/freeorion/default/python/AI/FreeOrionAI.py", line 330, in generateOrders
    action()
  File "/opt/projects/freeorion/default/python/AI/MilitaryAI.py", line 803, in assign_military_fleets_to_systems
    fleet_list=found_fleets)
  File "/opt/projects/freeorion/default/python/AI/FleetUtilsAI.py", line 150, in get_fleets_for_mission
    fleet_pool_set.remove(fleet_id)
KeyError: 114698
```

**Comment by MatGB** [comment]
OK, adding `
        EffectsGroup
            scope = And [
                Planet
                OwnedBy empire = Source.Owner
                Turn high = LocalCandidate.System.LastTurnBattleHere + 1
            ]
            priority = [[HIGH_PRIORITY]]
            effects = [
                SetConstruction value = Value - 2
                ]`
as a hack into DEF_PLAN_BARRIER_SHLD_1 works, as far as I can see the timings for when certain meters are changed has moved.

Probably related, if you have a colony building completed on an outpost, the human icon displays in the population stat before the effects have kicked off, I have vague memories of this happening while we were testing the buildings and it was squashed: if it's not related then a separate issue but I'm pretty sure it's caused by the same backend change.
![screenshot_20171006_075642](https://user-images.githubusercontent.com/11533206/31266529-1714429c-aa6c-11e7-983f-5a2120f316c9.png)

**Comment by MatGB** [comment]
@dbenage-cx yes, it was Mier y, sorry, for some reason Github didn't display your comment until I posted the above, if you can't replicate then there might be something wrong with my working branch given we've not had further reports, I might need to create a new one from master and cherry pick things, it's very weird though.

**Comment by dbenage-cx** [comment]
> the human icon displays in the population stat before the effects have kicked off

IIRC this happens when a planet has positive population but no species.  It does hint more towards a backend issue.

Can you reproduce the invasion issue with master?

**Comment by MatGB** [comment]
Yes, master as per above commit—don't want to recompile from current as this game has turned into a good test. It even shows up on immediate load, if I stash my above hack and load I can't invade a planet I can invade if my hack is running. I don't recall a bug that worked like that before you normally have to hit end turn.

Anyway, I've recreated my branch from master, cherry picked it in and finally created the PR, branch is here [research-tree-2](https://github.com/MatGB/freeorion/tree/research-tree-2), like I said I see nothing that should affect this, it's PR #1787. The bug is still there in the new tree, the old branch and master.

**Comment by Dilvish-fo** [comment]
> On load of save from turn 129, there is an AI exception, but doubt it is relevant as the planet shows accounting for PBS1:

That key error there is something that I only even notice on save-file load, but strangely seems to be related to some problems with the fleet-splitting operation not leaving the expected ship behind in the original fleet-- I had started investigating that at one point and then got pulled into other things before finishing off the bug, I'll try to get back to that soon.

**Comment by MatGB** [comment]
Had an unrelated crash recur so did a force reset of my master branch and recompiled as per 96e15f7, problem still there unless I apply the hack. Quitting, hitting git stash apply then loading allows me to invade immediately and succesfully, so appears to be a client side timing issue?

**Comment by dbenage-cx** [comment]
@Dilvish-fo Since there are no python changes in that branch, I opened a new issue just in case.


> so appears to be a client side timing issue?

But only for certain platforms? Posting shot of planet shield tooltip with debug dump info to see if anything else differs.
![screenshot_20171007_103523](https://user-images.githubusercontent.com/17205519/31309510-d2781bfe-ab4c-11e7-9054-7b60efbdfacb.png)

**Comment by MatGB** [comment]
OK, 2 things, one is an unrelated bug, I now have no planet stat tooltips displaying as per the most recent compile so can't check to see what techs are applied.

Second is my above hack was working because I used 'HIGH_PRIORITY' instead of EARLY_, which meant the entire tech file wasn't loading and the techs weren't working, so it was only working by happenstance.

I'll look at completing this game and then get a completely new copy of master and force a clean compile (again) to see if it's still replicated.

**Comment by MatGB** [comment]
OK, three more games completed, two in pure master and one in a new copy of my branch and everything is working fine. Weird edge case glitch I guess, thanks for the help guys.
